runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts

```diff
@@ -116,15 +115,66 @@ export default class ExcelActionHandler extends StatefulActionHandler {
         this.switchPrevWorksheet();
     }
 
+    async createWorksheet(
+        input: ExcelCreateWorksheetActionInput
+    ): Promise<ExcelCreateWorksheetActionOutput> {
+        const worksheets = this.session.Worksheets;
+        const worksheetsCount = worksheets.Count;
+        let worksheet: string;
+
+        if (!input.activate)
+            this.setPrevWorksheet(this.session.ActiveSheet.Name);
+
+        if (input.name) {
+            if (this.checkIfWorksheetExist(input.name))
+                throw new Error("Worksheet name already exists.");
+
+            worksheet = this.session.Worksheets.Add(
+                null,
+                this.session.Worksheets(worksheetsCount)
+            ).Name = input.name;
+        } else {
+            worksheet = this.session.Worksheets.Add(
+                null,
+                this.session.Worksheets(worksheetsCount)
+            ).Name;
```

Comment:
there's the same code used twice
also it's better to avoid let if possible
what do you think of 
const worksheet = input?.name 
? this.session.Worksheets.Add(
    null,
    this.session.Worksheets(worksheetsCount)
).Name = input.name
: this.session.Worksheets.Add(
    null,
    this.session.Worksheets(worksheetsCount)
).Name

or 
const worksheet = this.session.Worksheets.Add(
    null,
    this.session.Worksheets(worksheetsCount)
)
if(input?.name) <excel method that sets worksheet name - sth like: this.session.Worksheets(worksheet.Name).SetName(input?.name)

also replace this.session with const session = getSession()

Reply:
I think I'll leave it like this in this case due to that additional equals sign complicating things, but overall a great idea